### ISSUE TITLE
php 8.2 generates a lot of deprecation

### ISSUE THREAD

**Issue by giuliopons** [giuliopons/adadmin#77]
AdAdmin in not yet compatible with PHP 8.2

**Comment by shaahrokh** [comment]
As well with PHP 8.1
 
`Deprecated: Optional parameter $valore declared before required parameter $arrayvalori is implicitly treated as a required parameter in ../src/_include/formcampi.class.php on line 646`

**Comment by giuliopons** [comment]
> As well with PHP 8.1
> 
> `Deprecated: Optional parameter $valore declared before required parameter $arrayvalori is implicitly treated as a required parameter in ../src/_include/formcampi.class.php on line 646`

Can you tell me your AdAdmin version? I think I've already solved that problem.

**Comment by shaahrokh** [comment]
You're right. unfortunately our AdAdmin version is 3.98! Is there any way or document in order to update to latest version, because when I use the latest version of files I faced to schema error for database!

**Comment by giuliopons** [comment]
Here it is:
https://github.com/giuliopons/adadmin/discussions/32#discussioncomment-4028433

When you update the database should be automatically updated. Have you tried?
Giulio

**Comment by shaahrokh** [comment]
I tried but I have an error about database:

> Error executing query:
> 
> ALTER TABLE `7banner` CHANGE `dt_maxday_date` `dt_maxday_date` DATE NOT NULL DEFAULT '1970-01-01';
> .
> Error:
> Invalid default value for 'dt_giorno1'

**Comment by shaahrokh** [comment]
I did ti but I faced to these errors:
```
Error executing query:
ALTER TABLE `7banner` CHANGE `dt_maxday_date` `dt_maxday_date` DATE NOT NULL DEFAULT '1970-01-01';
.

Error:

Invalid default value for 'dt_giorno1'
```

**Comment by giuliopons** [comment]
Yes.
Great.

Once I used zero dates, but on some servers I noticed that they aren't allowed, so I've managed to change them to a default old date (such as 1970-01-01) but doing this changes sometimes doesn't work. Normally when these problems happen I ask for database credentials and manually fix the db.
You did the same. Well done.

I'm working to fix PHP 8.2 warnings!

**Comment by giuliopons** [comment]
The new version 4.2.4 should be compatible with PHP 8.2
Please check it! :-)
